src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/Streaming/DataStreamWriterTests.cs

```diff
@@ -86,13 +88,17 @@ public void TestForeachBatch()
                 .WriteStream()
                 .ForeachBatch((df, id) =>
                 {
-                    df.Write().Csv(Path.Combine(dstTempDirectory.Path, id.ToString()));
+                    Func<Column, Column> innerUdf = Udf<int, int>(i => i + 200);
```

Comment:
This won't work until the bug re broadcast variables is fixed.

Reply:
Can you explain why this is related to the broadcast variable bug?

Reply:
[s_jvmBroadcastVariables ](https://github.com/dotnet/spark/blob/ddafcfbf261e5b45adf0cde7579e1100b99fdfa8/src/csharp/Microsoft.Spark/Broadcast.cs#L265) is `[ThreadStatic]` and is `null` when the `ForeachBatchCallbackHandler` is called (since it runs in a separate worker task.

Reply:
oh ok. it's in a critical path (`CreatePythonFunction`). :)